src/midi.js

```diff
@@ -5,6 +5,68 @@ const assert = require('assert');
 const jsmidgen = require('jsmidgen');
 const setMiddleC = require('./setMiddleC');
 
+function doesArrayContainAnArray(arr){
+  return arr.some(function(ele){
+    return Object.prototype.toString.call( ele ) === '[object Array]'
+  });
+}
+
+/**
+ * Takes an array of clips and merges them on the same track. 
+ * @param  {Array} clipArray	Format: [{note: ['c3'], level: 127, length: 64}, [...], ...]
+ */
+const mergeMidiClips = (clipArray) => {
+
+	let clip = [];
+	let maxLenght = 0;
+	// Get the length of the final clip to be returned. (longest clip)
+	clipArray.forEach((noteObj) => {
+		if (noteObj.length > maxLenght) {
+			maxLenght = noteObj.length;
+		} 
+	});
+	// loop through all clicks and get the value of each clip.
+	let finalClip = [];
+	for(let i = 0; i < maxLenght; i++) {
+		let notes = {
+			note: null,
+			length: 0,
+			level: 0
+		};
+		// Loop through each clip and compile it into the finalClip.
+		clipArray.forEach((clipObj) => {	
+			// This loop is for when you have a uneven clip that doesn't reach the maxLenght. 
+			// So what it does is double smaller clips until it reaches target length.
+			let index = i;
+			while (clipObj[index] === undefined) {
```

Comment:
We have a bunch of other places where we try to double up arrays to match a particular length, maybe we can abstract it some time later. For now, you can just follow the same pattern as the other places in this repo:

```
if (clipArr.length < maxLength) {
	while (clipArr.length < maxLength) {
		clipArr = clipArr.concat(clipArr);
	}
	// Clip off extra notes
	clipArr = clipArr.slice(0, maxLength);
}
```